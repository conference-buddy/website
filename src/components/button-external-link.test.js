import React from 'react'
import { shallow } from 'enzyme'
import ButtonExternalLink from './button-external-link'

const testUrl = 'http://forum.conferencebuddy.io/'

const linkText = 'Test Button Linktext'
const linkImage = <img src="test.jpg" />

const requiredProps = {
  url: testUrl,
  children: linkText,
}

describe('<ButtonExternalLink />', () => {
  it('renders without crashing', () => {
    shallow(
      <ButtonExternalLink {...requiredProps}>{linkText}</ButtonExternalLink>
    )
  })

  describe('renders a link with an url and the "btn" class', () => {
    const wrapper = shallow(
      <ButtonExternalLink url={testUrl}>{linkText}</ButtonExternalLink>
    )
    it('renders a link', () => {
      expect(wrapper.find('a')).toHaveLength(1)
    })
    it('renders a link with a url', () => {
      expect(wrapper.find('a').prop('href')).toEqual(testUrl)
    })
    it('renders a link with a class', () => {
      expect(wrapper.hasClass('btn')).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('renders an a-element with a given content', () => {
    it('renders a link with a text as children', () => {
      const wrapper = shallow(
        <ButtonExternalLink url={testUrl}>{linkText}</ButtonExternalLink>
      )
      expect(wrapper.text()).toBe(linkText)
      expect(wrapper).toMatchSnapshot()
    })
    it('renders a link with an image as children', () => {
      const wrapper = shallow(
        <ButtonExternalLink url={testUrl}>{linkImage}</ButtonExternalLink>
      )
      expect(wrapper.contains(linkImage)).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('adds a additional class dependend on props', () => {
    it('adds a class for style primary', () => {
      const wrapper = shallow(
        <ButtonExternalLink url={testUrl} primary>
          {linkText}
        </ButtonExternalLink>
      )
      expect(wrapper.hasClass('btn-primary')).toBe(true)
      expect(wrapper.hasClass('btn-secondary')).toBe(false)
      expect(wrapper).toMatchSnapshot()
    })
    it('adds a class for style secondary', () => {
      const wrapper = shallow(
        <ButtonExternalLink url={testUrl} secondary>
          {linkText}
        </ButtonExternalLink>
      )
      expect(wrapper.hasClass('btn-secondary')).toBe(true)
      expect(wrapper.hasClass('btn-primary')).toBe(false)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('adds a additional class dependend on props', () => {
    const wrapperSmallBtn = shallow(
      <ButtonExternalLink url={testUrl} small>
        {linkText}
      </ButtonExternalLink>
    )
    const wrapperBtn = shallow(
      <ButtonExternalLink url={testUrl}>{linkText}</ButtonExternalLink>
    )
    it('adds "btn-custom" class if no small prop is given', () => {
      expect(wrapperBtn.hasClass('btn-custom')).toBe(true)
      expect(wrapperBtn.hasClass('btn-custom-small')).toBe(false)
      expect(wrapperBtn).toMatchSnapshot()
    })
    it('adds "btn-custom-small" class if prop small is true', () => {
      expect(wrapperSmallBtn.hasClass('btn-custom')).toBe(false)
      expect(wrapperSmallBtn.hasClass('btn-custom-small')).toBe(true)
      expect(wrapperSmallBtn).toMatchSnapshot()
    })
  })
})
