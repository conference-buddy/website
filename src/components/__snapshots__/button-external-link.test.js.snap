// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonExternalLink /> adds a additional class dependend on props adds "btn-custom" class if no small prop is given 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonExternalLink
    url="http://forum.conferencebuddy.io/"
  >
    Test Button Linktext
  </ButtonExternalLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Test Button Linktext",
      "className": "btn btn-custom",
      "href": "http://forum.conferencebuddy.io/",
    },
    "ref": null,
    "rendered": "Test Button Linktext",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test Button Linktext",
        "className": "btn btn-custom",
        "href": "http://forum.conferencebuddy.io/",
      },
      "ref": null,
      "rendered": "Test Button Linktext",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ButtonExternalLink /> adds a additional class dependend on props adds "btn-custom-small" class if prop small is true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonExternalLink
    small={true}
    url="http://forum.conferencebuddy.io/"
  >
    Test Button Linktext
  </ButtonExternalLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Test Button Linktext",
      "className": "btn btn-custom-small",
      "href": "http://forum.conferencebuddy.io/",
    },
    "ref": null,
    "rendered": "Test Button Linktext",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test Button Linktext",
        "className": "btn btn-custom-small",
        "href": "http://forum.conferencebuddy.io/",
      },
      "ref": null,
      "rendered": "Test Button Linktext",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ButtonExternalLink /> adds a additional class dependend on props adds a class for style primary 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonExternalLink
    primary={true}
    url="http://forum.conferencebuddy.io/"
  >
    Test Button Linktext
  </ButtonExternalLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Test Button Linktext",
      "className": "btn btn-custom btn-primary",
      "href": "http://forum.conferencebuddy.io/",
    },
    "ref": null,
    "rendered": "Test Button Linktext",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test Button Linktext",
        "className": "btn btn-custom btn-primary",
        "href": "http://forum.conferencebuddy.io/",
      },
      "ref": null,
      "rendered": "Test Button Linktext",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ButtonExternalLink /> adds a additional class dependend on props adds a class for style secondary 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonExternalLink
    secondary={true}
    url="http://forum.conferencebuddy.io/"
  >
    Test Button Linktext
  </ButtonExternalLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Test Button Linktext",
      "className": "btn btn-custom btn-secondary",
      "href": "http://forum.conferencebuddy.io/",
    },
    "ref": null,
    "rendered": "Test Button Linktext",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test Button Linktext",
        "className": "btn btn-custom btn-secondary",
        "href": "http://forum.conferencebuddy.io/",
      },
      "ref": null,
      "rendered": "Test Button Linktext",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ButtonExternalLink /> renders a link with an url and the "btn" class renders a link with a class 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonExternalLink
    url="http://forum.conferencebuddy.io/"
  >
    Test Button Linktext
  </ButtonExternalLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Test Button Linktext",
      "className": "btn btn-custom",
      "href": "http://forum.conferencebuddy.io/",
    },
    "ref": null,
    "rendered": "Test Button Linktext",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test Button Linktext",
        "className": "btn btn-custom",
        "href": "http://forum.conferencebuddy.io/",
      },
      "ref": null,
      "rendered": "Test Button Linktext",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ButtonExternalLink /> renders an a-element with a given content renders a link with a text as children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonExternalLink
    url="http://forum.conferencebuddy.io/"
  >
    Test Button Linktext
  </ButtonExternalLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Test Button Linktext",
      "className": "btn btn-custom",
      "href": "http://forum.conferencebuddy.io/",
    },
    "ref": null,
    "rendered": "Test Button Linktext",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test Button Linktext",
        "className": "btn btn-custom",
        "href": "http://forum.conferencebuddy.io/",
      },
      "ref": null,
      "rendered": "Test Button Linktext",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ButtonExternalLink /> renders an a-element with a given content renders a link with an image as children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonExternalLink
    url="http://forum.conferencebuddy.io/"
  >
    <img
      src="test.jpg"
    />
  </ButtonExternalLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        src="test.jpg"
      />,
      "className": "btn btn-custom",
      "href": "http://forum.conferencebuddy.io/",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "src": "test.jpg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          src="test.jpg"
        />,
        "className": "btn btn-custom",
        "href": "http://forum.conferencebuddy.io/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "test.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
